problem_report.py

```diff
@@ -550,6 +522,40 @@ def write(self, file, only_new=False):
                 file.write(base64.b64encode(block))
                 file.write(b"\n")
 
+    def sort_keys(self, only_new):
```

Comment:
suggestion: Can you add type hints to the new function? I am doing that for new code to apport.

```suggestion
    def sort_keys(self, only_new: bool) -> typing.Tuple[list[str], list[str]]:
```

Reply:
I can do that.

I dislike them, as I think they are misleading, because the type checks that can be done don't guarantee anything at execution time.

But if that is what it takes, I can do that.

Reply:
I see two benefits of them: Added documentation for the caller and finding some (but not all) type issues. We will run mypy on the code find them (see https://github.com/canonical/apport/pull/184). We found 2-3 real bugs thanks to mypy.